tests/saw/bike_r2/proof/aes.saw

```diff
@@ -0,0 +1,198 @@
+      memory errors.  So this invariant is an important property to track.
+*/
+
+let init_aes_ctr_prf_state_spec = do {
+  s_ptr <- crucible_alloc aes_ctr_prf_state_t;
+  max_invokations <- crucible_fresh_var "max_invokations" i32;
+  (_, seed_ptr) <- in_ref seed_t "seed";
+
+  crucible_precond {{ max_invokations > 0 }};
+
+  crucible_execute_func [s_ptr, tm max_invokations, seed_ptr];
+
+  ctx_ptr <- crucible_alloc evp_cipher_ctx_st_t;
+  s <- points_to_fresh_aes_ctr_prf_state s_ptr ctx_ptr "s";
+
+  crucible_postcond {{ s.rem_invokations == max_invokations }};
+  crucible_postcond {{ s.pos == `AES256_BLOCK_SIZE }};
+
+  crucible_return (tm {{fromInteger`{[32]} SUCCESS}});
+};
+
+let aes_ctr_prf_spec len check_rem_invokations = do {
+  a_ptr <- out_ref (i8_array_t len);
+  (s, s_ptr, ctx_ptr) <- ptr_to_fresh_aes_ctr_prf_state "s";
+
+  crucible_precond {{ s.pos <= `AES256_BLOCK_SIZE }};
+
+  if check_rem_invokations then do {
+    crucible_precond {{ s.rem_invokations >= `(len / AES256_BLOCK_SIZE) + 1 }};
+  } else do {
+    return ();
+  };
+
+  crucible_precond {{ `len < 2^^32 - `AES256_BLOCK_SIZE }};
+
+  crucible_execute_func [a_ptr, s_ptr, tm {{ `len : [32] }}];
+
+  _ <- point_to (i8_array_t len) a_ptr "a";
+  s' <- points_to_fresh_aes_ctr_prf_state s_ptr ctx_ptr "s'";
+
+  crucible_postcond {{ s'.pos <= `AES256_BLOCK_SIZE }};
+
+  if check_rem_invokations then do {
+    crucible_postcond {{ s'.rem_invokations >= s.rem_invokations - `(len / AES256_BLOCK_SIZE) - 1 }};
+  } else do {
+    return ();
+  };
+
+  crucible_return (tm {{fromInteger`{[32]} SUCCESS}});
+};
+
+let finalize_aes_ctr_prf_spec = do {
+  (_, s_ptr, _) <- ptr_to_fresh_aes_ctr_prf_state "s";
+  crucible_execute_func [s_ptr];
+};
+
+
+///////////////////////////////////////////////////////////////////////////////
+// Proof Commands
+
+EVP_CIPHER_CTX_new_ov <- admit "EVP_CIPHER_CTX_new" [] EVP_CIPHER_CTX_new_spec;
+
+EVP_CIPHER_CTX_free_ov <- admit "EVP_CIPHER_CTX_free" [] EVP_CIPHER_CTX_free_spec;
+
+EVP_CIPHER_CTX_set_padding_ov <- admit "EVP_CIPHER_CTX_set_padding" [] EVP_CIPHER_CTX_set_padding_spec;
+
+EVP_EncryptInit_ex_ov <- admit "EVP_EncryptInit_ex" [] EVP_EncryptInit_ex_spec;
+
+EVP_EncryptUpdate_ov <- admit "EVP_EncryptUpdate" [] EVP_EncryptUpdate_spec;
+
+EVP_aes_256_ecb_ov <- admit "EVP_aes_256_ecb" [] EVP_aes_256_ecb_spec;
+
+aes256_key_expansion_ov <- verify "aes256_key_expansion"
+  [ EVP_CIPHER_CTX_new_ov
+  , EVP_CIPHER_CTX_free_ov
+  , EVP_CIPHER_CTX_set_padding_ov
+  , EVP_EncryptInit_ex_ov
+  , EVP_aes_256_ecb_ov
+  ]
+  aes256_key_expansion_spec;
+
+aes256_enc_ov <- verify "aes256_enc"
+  [EVP_EncryptUpdate_ov]
+  aes256_enc_spec;
+
+aes256_free_ks_ov <- verify "aes256_free_ks"
+  [EVP_CIPHER_CTX_free_ov]
+  aes256_free_ks_spec;
+
+init_aes_ctr_prf_state_ov <- verify init_aes_ctr_prf_state_fun_name
+  [aes256_key_expansion_ov]
+  init_aes_ctr_prf_state_spec;
+
+aes_ctr_prf_4_ov <- admit aes_ctr_prf_fun_name
```

Comment:
I noticed this one has an override. Is this an intentional admit?

Reply:
this is `admit` because the spec is actually incorrect, in that the call succeeds regardless of the value of `rem_invokations`. This is used by `get_rand_mod_len` which could in run out of invocations and fail, but we can't write the correct spec in `saw`.